R/translator.R

```diff
@@ -0,0 +1,108 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    translation_language = "character",
+    options = "list",
+    translations = "data.frame",
+    mode = "character"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_csvs_path = NULL,
+                        translation_json_path = NULL,
+                        translation_csv_config = NULL) {
+    options <<- .translator_options
+    # Reading translations.
+    if (!is.null(translation_csvs_path))
+      .read_csv(translation_csvs_path, translation_csv_config)
+    else if (!is.null(translation_json_path))
+      .read_json(translation_json_path)
+    else
+      stop("You must provide either translation json or csv files.")
+    # Set default translation language
```

Comment:
this comment is mysterious to me. is `character(0)` really the _default language_? is it a language at all? what I understand from the code is that it is a marker that the language is not set, but the comment is saying something different. I'd prefer not to have a comment at all, especially since this line seems pretty straightforward